### PR TITLE
Fix coverageReport gradle task

### PR DESCRIPTION
**Pull request by russelarms**
Please have a look at our [guidelines for contributions](https://github.com/microsoft/appcenter-sdk-android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] ~~Has `CHANGELOG.md` been updated?~~
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] ~~Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?~~
* [ ] ~~Did you check UI tests on the sample app? They are not executed on CI.~~

## Description

The code coverage has been broken on Android since we bumped up the Gradle plugin version. 
This PR:
- Fixes the code coverage task
- Adds and changes tests to reach 100% coverage
- Removes unused methods (postOnUiThread & runIfEnabled)

## Related PRs or issues

[AB#84105](https://msmobilecenter.visualstudio.com/Mobile-Center/_boards/board/t/Ivan-Team/Backlog%20Items/?workitem=84105)

### CODE REVIEW

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeCustomizationTest.java [hunk 1]
```diff
@@ -172,16 +174,79 @@ public void distributeListenerNoReleaseFound() throws Exception {
...
+    @Test
+    public void distributeNoActivity() throws Exception {
+        System.out.println("3");
+        DistributeListener listener = mock(DistributeListener.class);
+        distributeNoReleaseAvailableCoverage(listener, listener, true);
+    }
+
+    @Test
+    public void distributeNoListenerNoActivity() throws Exception {
+        System.out.println("4");
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeCustomizationTest.java:205]: Why we need this line?

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeCustomizationTest.java [hunk 2]
```diff
@@ -172,16 +174,79 @@ public void distributeListenerNoReleaseFound() throws Exception {
...
+        DistributeListener listener = mock(DistributeListener.class);
+        distributeNoReleaseAvailableCoverage(null, listener, true);
+    }
+
+    ServiceCallback callback = null;
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeCustomizationTest.java:210]: Let's move it to top

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeCustomizationTest.java [hunk 3]
```diff
@@ -172,16 +174,79 @@ public void distributeListenerNoReleaseFound() throws Exception {
...
+
+        /* Set Distribute listener and customize it. */
+        Distribute.setListener(actualListener);
+
+        /* Resume activity. */
+        Distribute.getInstance().onActivityResumed(mActivity);
+        if (onPause){
+            Distribute.getInstance().onActivityPaused(mActivity);
+        }
+
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeCustomizationTest.java:244]: extra line

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeCustomizationTest.java [hunk 4]
```diff
@@ -172,16 +174,79 @@ public void distributeListenerNoReleaseFound() throws Exception {
...
         /* Resume activity again to invoke update request. */
-        Distribute.getInstance().onActivityPaused(mActivity);
         Distribute.getInstance().onActivityResumed(mActivity);
 
         /* Verify the right listener gets called again. */
-        verify(listener).onNoReleaseAvailable(mActivity);
-        verify(listener, times(0)).onReleaseAvailable(any(Activity.class), any(ReleaseDetails.class));
+        verify(listener, times(1)).onNoReleaseAvailable(mActivity);
+        verify(listener, never()).onReleaseAvailable(any(Activity.class), any(ReleaseDetails.class));
+
```
Comment (AnastasiaKubova) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeCustomizationTest.java:186]: extra line

File: sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/utils/ErrorLogHelper.java [hunk 5]
```diff
@@ -470,7 +470,7 @@ public static ErrorReport getErrorReportFromErrorLog(@NonNull ManagedErrorLog lo
     }
 
     @VisibleForTesting
```
Comment (MatkovIvan) [inline sdk/appcenter-crashes/src/main/java/com/microsoft/appcenter/crashes/utils/ErrorLogHelper.java:472]: It's not required if it's public

File: sdk/appcenter-crashes/src/test/java/com/microsoft/appcenter/crashes/CrashesTest.java [hunk 6]
```diff
@@ -1565,6 +1564,46 @@ public void checkStacktraceWhenThrowableFileIsNull() throws Exception {
...
+        mockLog.setDevice(new Device());
+
+        /* Build error report. */
+        Crashes crashes = Mockito.spy(Crashes.getInstance());
+        ErrorReport report = crashes.buildErrorReport(mockLog);
+
+        /* Verify. */
+        assertEquals(expectedStacktrace, report.getStackTrace());
+        verify(crashes, never()).buildStackTrace(any(com.microsoft.appcenter.crashes.ingestion.models.Exception.class));
+        verifyStatic(times(1));
```
Comment (MatkovIvan) [inline sdk/appcenter-crashes/src/test/java/com/microsoft/appcenter/crashes/CrashesTest.java:1603]:
```suggestion
        verifyStatic();
```

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeCustomizationTest.java [hunk 7]
```diff
@@ -172,16 +174,129 @@ public void distributeListenerNoReleaseFound() throws Exception {
...
+        verify(listener, never()).onReleaseAvailable(any(Activity.class), any(ReleaseDetails.class));
 
         /* Resume activity again to invoke update request. */
-        Distribute.getInstance().onActivityPaused(mActivity);
         Distribute.getInstance().onActivityResumed(mActivity);
 
         /* Verify the right listener gets called again. */
-        verify(listener).onNoReleaseAvailable(mActivity);
-        verify(listener, times(0)).onReleaseAvailable(any(Activity.class), any(ReleaseDetails.class));
+        verify(listener, times(1)).onNoReleaseAvailable(mActivity);
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeCustomizationTest.java:184]:
```suggestion
        verify(listener).onNoReleaseAvailable(mActivity);
```

**Review by MatkovIvan — APPROVED** [review]

**Review by Jamminroot — APPROVED** [review]

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/AbstractAppCenterService.java [hunk 8]
```diff
@@ -355,56 +354,4 @@ public void run() {
...
-    /**
-     * Like {{@link #post(Runnable)}} but also post back in U.I. thread.
-     * Use this for example to manage life cycle callbacks to make sure SDK is started and that
-     * every operation runs in order.
-     * <p>
-     * This method will not run the command if the SDK is disabled, the purpose is for internal commands, not APIs.
-     *
-     * @param runnable command to run.
-     */
-    protected synchronized void postOnUiThread(final Runnable runnable) {
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/AbstractAppCenterService.java:368]: Was that used only in distribute then removed?

Comment (russelarms) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/AbstractAppCenterService.java:368]: It was used only in Push.

**Review by AnastasiaKubova — APPROVED** [review]

**Pull request merged**